source/includes/monitoring/monitoring.rb

```diff
@@ -0,0 +1,173 @@
+topology_changed_subscriber = TopologyChangedLogSubscriber.new
+server_closed_subscriber = ServerClosedLogSubscriber.new
+topology_closed_subscriber = TopologyClosedLogSubscriber.new
+
+Mongo::Monitoring::Global.subscribe(Mongo::Monitoring::TOPOLOGY_OPENING,
+  topology_opening_subscriber)
+Mongo::Monitoring::Global.subscribe(Mongo::Monitoring::SERVER_OPENING,
+  server_opening_subscriber)
+Mongo::Monitoring::Global.subscribe(Mongo::Monitoring::SERVER_DESCRIPTION_CHANGED,
+  server_description_changed_subscriber)
+Mongo::Monitoring::Global.subscribe(Mongo::Monitoring::TOPOLOGY_CHANGED,
+  topology_changed_subscriber)
+Mongo::Monitoring::Global.subscribe(Mongo::Monitoring::SERVER_CLOSED,
+  server_closed_subscriber)
+Mongo::Monitoring::Global.subscribe(Mongo::Monitoring::TOPOLOGY_CLOSED,
+  topology_closed_subscriber)
+# end-sdam-subscriber-global
+
+# start-sdam-subscriber-client
+topology_opening_subscriber = TopologyOpeningLogSubscriber.new
+server_opening_subscriber = ServerOpeningLogSubscriber.new
+server_description_changed_subscriber = ServerDescriptionChangedLogSubscriber.new
+topology_changed_subscriber = TopologyChangedLogSubscriber.new
+server_closed_subscriber = ServerClosedLogSubscriber.new
+topology_closed_subscriber = TopologyClosedLogSubscriber.new
+
+sdam_proc = Proc.new do |client|
+  client.subscribe(Mongo::Monitoring::TOPOLOGY_OPENING,
+    topology_opening_subscriber)
+  client.subscribe(Mongo::Monitoring::SERVER_OPENING,
+    server_opening_subscriber)
+  client.subscribe(Mongo::Monitoring::SERVER_DESCRIPTION_CHANGED,
+    server_description_changed_subscriber)
+  client.subscribe(Mongo::Monitoring::TOPOLOGY_CHANGED,
+    topology_changed_subscriber)
+  client.subscribe(Mongo::Monitoring::SERVER_CLOSED,
+    server_closed_subscriber)
+  client.subscribe(Mongo::Monitoring::TOPOLOGY_CLOSED,
+    topology_closed_subscriber)
+end
+
+client = Mongo::Client.new(['127.0.0.1:27017'], database: 'test',
+  sdam_proc: sdam_proc)
+# end-sdam-subscriber-client
+
+# start-heartbeat
+class HeartbeatLogSubscriber
+    include Mongo::Loggable
+  
+    def started(event)
+      log_debug("#{event.address} | STARTED")
+    end
+  
+    def succeeded(event)
+      log_debug("#{event.address} | SUCCEEDED | #{event.duration}s")
+    end
+  
+    def failed(event)
+      log_debug("#{event.address} | FAILED | #{event.error.class}: #{event.error.message} | #{event.duration}s")
+    end
+  
+    private
+  
+    def logger
+      Mongo::Logger.logger
+    end
+  
+    def format_message(message)
+      format("HEARTBEAT | %s".freeze, message)
+    end
+  end
+# end-heartbeat
+
+# start-heartbeat-subscribe
+subscriber = HeartbeatLogSubscriber.new
+
+# Globally subscribes to Server Opening events
+Mongo::Monitoring::Global.subscribe(Mongo::Monitoring::SERVER_HEARTBEAT, subscriber)
+
+# Subscribes to Server Opening events at the client level
+client = Mongo::Client.new([ '127.0.0.1:27017' ], :database => 'test' )
```

Comment:
Prefer `symbol: 'value'` syntax for hashes and named arguments whenever possible:

```suggestion
client = Mongo::Client.new([ '127.0.0.1:27017' ], database: 'test' )
```